File: exercism/go/scrabble-score/scrabble_score.go
package scrabble

import "unicode"

func Score(word string) (res int) {
	for _, c := range word {
		switch unicode.ToLower(c) {
		case 'a', 'e', 'i', 'o', 'u', 'l', 'n', 'r', 's', 't':
			res += 1
		case 'd', 'g':
			res += 2
		case 'b', 'c', 'm', 'p':
			res += 3
		case 'f', 'h', 'v', 'w', 'y':
			res += 4
		case 'k':
			res += 5
		case 'j', 'x':
			res += 8
		case 'q', 'z':
			res += 10
		default:
			panic("unrecognized letter")
		}
	}
	return res
}

File: exercism/zig/scrabble-score/scrabble_score.zig
const toUpper = @import("std").ascii.toUpper;

pub fn score(s: []const u8) u32 {
    var res: u32 = 0;
    for (s) |letter| {
        res += switch (toUpper(letter)) {
            'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' => 1,
            'D', 'G' => 2,
            'B', 'C', 'M', 'P' => 3,
            'F', 'H', 'V', 'W', 'Y' => 4,
            'K' => 5,
            'J', 'X' => 8,
            'Q', 'Z' => 10,
            else => @panic("unexpected character" ++ .{letter}),
        };
    }
    return res;
}
